cmake/catkin_simple-extras.cmake.em

```diff
@@ -109,6 +109,21 @@ macro(catkin_simple)
       list(INSERT ${PROJECT_NAME}_CATKIN_BUILD_DEPENDS_EXPORTED_TARGETS 0 ${${PROJECT_NAME}_EXPORTED_TARGETS})
     endif()
   endif()
+  
+  # generate dynamic reconfigure files
+  if(dynamic_reconfigure_FOUND_CATKIN_PROJECT)
+    set(${PROJECT_NAME}_LOCAL_CFG_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cfg)
+    if(IS_DIRECTORY ${${PROJECT_NAME}_LOCAL_CFG_DIR})
+        # create a list containing all the cfg files
+        file(GLOB ${PROJECT_NAME}_LOCAL_CFG_FILES RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "${${PROJECT_NAME}_LOCAL_CFG_DIR}/*.cfg")
+        # process each cfg file if the list is not empty
+        if(${PROJECT_NAME}_LOCAL_CFG_FILES)
+            foreach(cfg_file ${${PROJECT_NAME}_LOCAL_CFG_FILES})
+                generate_dynamic_reconfigure_options(${cfg_file})
```

Comment:
I think you should attempt to pass all the files to one invocation of `generate_dynamic_reconfigure_options` as the tutorial suggests:

http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile#dynamic_reconfigure.2BAC8-Tutorials.2BAC8-HowToWriteYourFirstCfgFile.2BAC8-catkin.Use_the_cfg_File

Also, you may need to add dependencies on the `gencfg` target from the other targets built in this package to prevent build race conditions. This is also mentioned by the above link.